ivy/functional/frontends/paddle/nn/functional/activation.py

```diff
@@ -1,9 +1,69 @@
 # local
+import ivy
+from ivy.utils.exceptions import handle_exceptions
+from ivy.func_wrapper import with_supported_dtypes
+from ivy.functional.frontends.paddle.func_wrapper import to_ivy_arrays_and_back
 from ivy.functional.frontends.paddle.tensor.math import tanh as paddle_tanh
 from ivy.functional.frontends.paddle.tensor.math import (
     log_softmax as paddle_log_softmax,
 )
 
 
+def _batch_promotion(*args, default_dtype="float64"):
+    # Promote all types
+    promote_types = set()
+
+    for arg in args:
+        if args is None:
+            continue
+        if isinstance(arg, (float, int)):
+            continue
+        promote_types.add(ivy.dtype(arg))
+
+    if "float64" in promote_types:
+        return "float64"
+
+    if "float32" in promote_types:
+        return "float32"
+
+    if "float16" in promote_types:
+        return "float32" if "bfloat16" in promote_types else "float16"
+
+    if "bfloat16" in promote_types:
+        return "bfloat16"
+
+    if "int64" in promote_types or "uint64" in promote_types:
+        return "float64"
+
+    ints = ["int8", "int16", "int32"]
+    if "uint32" in promote_types and any(d in promote_types for d in ints):
+        return "float64"
+
+    return default_dtype
+
+
+@with_supported_dtypes({"2.4.2 and below": ("float32", "float64")}, "paddle")
+@to_ivy_arrays_and_back
+@handle_exceptions
```

Comment:
The `@handle_exceptions` decorator is used to apply to functions in the Functional API only since it raises either one of the `IvyNotImplementedException`, `IvyError` or `IvyBackendException`. Can you kindly remove this? :)